explorer-ui/src/components/ProfileView/Preferences/Preferences.jsx

```diff
@@ -41,36 +48,43 @@ function handleAttractionsSelected(event, attraction) {
   // Update the state array for the users attractions interests
 function handleOccupationsSelected(event, occupation) {
     event.preventDefault();
-    setEmployment(occupation)
+    setEmployment(occupation.current.value,)
     setPreferenceInfo({"interests": preferenceInfo.interests, "attractions": preferenceInfo.attractions, "occupation": employment, "budget": preferenceInfo.budget, "currLocation": preferenceInfo.currLocation, "visibility": preferenceInfo.visibility})
     console.log("occupation is: ", employment)
   };
 
   // Update the state array for the users attractions interests
   function handleBudgetSelected(event, selectedBudget) {
     event.preventDefault();
-    setBudget(selectedBudget)
+    setBudget(selectedBudget.current.value)
     setPreferenceInfo({"interests": preferenceInfo.interests, "attractions": preferenceInfo.attractions, "occupation": preferenceInfo.occupation, "budget": budget, "currLocation": preferenceInfo.currLocation, "visibility": preferenceInfo.visibility})
     console.log("budget is: ", budget)
   };
 
     // Update the state array for the users attractions interests
     function handleVisibilitySelected(event, visibililityChoice) {
         event.preventDefault();
-        setVisibility(visibililityChoice)
+        setVisibility(visibililityChoice.current.value)
         setPreferenceInfo({"interests": preferenceInfo.interests, "attractions": preferenceInfo.attractions, "occupation": preferenceInfo.occupation, "budget": preferenceInfo.budget, "currLocation": preferenceInfo.currLocation, "visibility": visibility})
         console.log("visibility is: ", visibililityChoice)
       };
+  // Update the state array for the users attractions interests
+  function handleLocationSet(event, country, province, city) {
+    event.preventDefault();
+    setCurrLocation({"country":country.current.value, "province":province.current.value, "city": city.current.value})
+    setPreferenceInfo({"interests": preferenceInfo.interests, "attractions": preferenceInfo.attractions, "occupation": preferenceInfo.employment, "budget": preferenceInfo.budget, "currLocation": currLocation, "visibility": preferenceInfo.visibility})
```

Comment:
for state updates - would be careful on ensuring updates have the most recent updated value i.e here currentLocation when setting preference info